### PR TITLE
Implemented orders api for opencart 2.3

### PR DESCRIPTION
**Pull request by mayankamencherla**

### CODE REVIEW

File: catalog/controller/extension/payment/razorpay.php [hunk 1]
```diff
@@ -1,15 +1,24 @@
...
+require_once __DIR__.'/../../razorpay-sdk/Razorpay.php';
+use Razorpay\Api\Api;
+
 class ControllerExtensionPaymentRazorpay extends Controller
 {
     public function index()
     {
         $data['button_confirm'] = $this->language->get('button_confirm');
-
+
```
Comment (73sl4) [inline catalog/controller/extension/payment/razorpay.php]: extra spaces.

File: catalog/controller/extension/payment/razorpay.php [hunk 2]
```diff
@@ -1,15 +1,24 @@
...
         $data['button_confirm'] = $this->language->get('button_confirm');
-
+        
         $this->load->model('checkout/order');
 
         $order_info = $this->model_checkout_order->getOrder($this->session->data['order_id']);
 
+        // Orders API with payment autocapture
+        $api = new Api($this->config->get('razorpay_key_id'), $this->config->get('razorpay_key_secret'));
+        $order_data = $this->get_order_creation_data($this->session->data['order_id']);
```
Comment (73sl4) [inline catalog/controller/extension/payment/razorpay.php]: line breaks

File: catalog/controller/payment/razorpay.php [hunk 3]
```diff
@@ -1,15 +1,24 @@
 <?php
 
+require_once __DIR__.'/../razorpay-sdk/Razorpay.php';
+use Razorpay\Api\Api;
+
 class ControllerPaymentRazorpay extends Controller
 {
-    public function index()
+    protected function index()
```
Comment (73sl4) [inline catalog/controller/payment/razorpay.php]: Why are changing this to protected.

File: catalog/controller/extension/payment/razorpay.php [hunk 4]
```diff
@@ -28,99 +38,121 @@ public function index()
...
+            {
+                if ($this->payment_action === 'authorize')
+                {   
+                    $payment = $api->payment->fetch($razorpay_payment_id);
+                }
+                else
+                {   
+                    $signature = hash_hmac('sha256', $razorpay_order_id . '|' . $razorpay_payment_id, $key_secret);
+
+                    if (hash_equals($signature , $razorpay_signature))
```
Comment (73sl4) [inline catalog/controller/extension/payment/razorpay.php]: Add check for hash_equals function exists.

Comment (mayankamencherla) [inline catalog/controller/extension/payment/razorpay.php]: added

File: catalog/controller/payment/razorpay.php (hunk 3 above)
Comment (mayankamencherla) [inline catalog/controller/payment/razorpay.php]: changed

File: catalog/controller/extension/payment/razorpay.php (hunk 2 above)
Comment (mayankamencherla) [inline catalog/controller/extension/payment/razorpay.php]: added

File: catalog/controller/extension/payment/razorpay.php (hunk 1 above)
Comment (mayankamencherla) [inline catalog/controller/extension/payment/razorpay.php]: fixed

Comment (73sl4) [inline catalog/controller/extension/payment/razorpay.php]: bump.

File: catalog/controller/extension/payment/razorpay.php (hunk 2 above)
Comment (73sl4) [inline catalog/controller/extension/payment/razorpay.php]: bump.

File: catalog/controller/extension/payment/razorpay.php [hunk 5]
```diff
@@ -28,99 +38,121 @@ public function index()
...
-        curl_setopt($ch, CURLOPT_POST, 1);
-        curl_setopt($ch, CURLOPT_POSTFIELDS, $fields_string);
-        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
-        curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, true);
-        curl_setopt($ch, CURLOPT_CAINFO, dirname(__FILE__).'/ca-bundle.crt');
-
-        return $ch;
+        $order = $this->model_checkout_order->getOrder($this->session->data['order_id']);
+        
+        switch($this->payment_action)
```
Comment (73sl4) [inline catalog/controller/extension/payment/razorpay.php]: Use ternary operator for adding payment_capture value.

File: catalog/controller/extension/payment/razorpay.php [hunk 6]
```diff
@@ -28,99 +38,121 @@ public function index()
...
-                    $result = curl_exec($ch);
-                $http_status = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+                //Check success response
+                if ($captured)
+                {
+                    $success = true;
+                }
 
-                if ($result === false) {
+                else{
```
Comment (73sl4) [inline catalog/controller/extension/payment/razorpay.php]: brackets

File: catalog/controller/extension/payment/razorpay.php (hunk 5 above)
Comment (mayankamencherla) [inline catalog/controller/extension/payment/razorpay.php]: added

File: catalog/controller/extension/payment/razorpay.php (hunk 6 above)
Comment (mayankamencherla) [inline catalog/controller/extension/payment/razorpay.php]: changed

**Comment by mayankamencherla** [comment]
@73sl4 Is the `is_serialized` method even needed?

**Pull request merged**